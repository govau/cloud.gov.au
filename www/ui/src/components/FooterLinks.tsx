import * as React from "react";
import styled from "../styled-components";

import BaseLink, { LinkProps, RouterLinkProps } from "./Link";

const Ul = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
`;

const Li = styled.li`
  display: inline-block;
  margin: 1rem 1rem 0 0;
`;

const UnstyledLink: React.StatelessComponent<
  Partial<LinkProps & RouterLinkProps>
> = ({ children, ...rest }) => <BaseLink {...rest}>{children}</BaseLink>;

const Link = styled(UnstyledLink)``;

interface Props {}

const FooterLinks: React.StatelessComponent<Props> = () => (
  <Ul>
    <Li>
      <Link href="https://console.system.b.cld.gov.au/">
        Production console
      </Link>
    </Li>
    <Li>
      <Link href="https://console.system.y.cld.gov.au/">Staging console</Link>
    </Li>
    <Li>
      <Link href="http://status.cloud.gov.au/">Status</Link>
    </Li>
  </Ul>
);

export default FooterLinks;
