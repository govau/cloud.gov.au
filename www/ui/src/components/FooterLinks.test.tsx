import * as React from "react";
import { MemoryRouter } from "react-router";
import * as renderer from "react-test-renderer";

import FooterLinks from "./FooterLinks";

it("renders without crashing", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <FooterLinks />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
