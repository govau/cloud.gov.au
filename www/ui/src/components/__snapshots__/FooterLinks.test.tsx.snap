// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c2 {
  -webkit-transition: 0.25s;
  transition: 0.25s;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 1px;
}

.c0 {
  margin: 0;
  padding: 0;
  list-style: none;
}

.c1 {
  display: inline-block;
  margin: 1rem 1rem 0 0;
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <a
      className="c2"
      href="https://console.system.b.cld.gov.au/"
    >
      Production console
    </a>
  </li>
  <li
    className="c1"
  >
    <a
      className="c2"
      href="https://console.system.y.cld.gov.au/"
    >
      Staging console
    </a>
  </li>
</ul>
`;
